### DISCUSSION TITLE
Base directory for media server community containers

### DISCUSSION THREAD

**Discussion by jo-me** [nextcloud/all-in-one#3939] [Questions] [answered]
Hi,
I stumbled over the community containers when looking for a solution to better integrate Nextcloud with Media servers such as Plex/Jellyfin and (to some extent) Memories. 

What the community container JSONs all ahve in common is the %NEXTCLOUD_DATADIR% as main mountpoint for media.
(see https://github.com/nextcloud/all-in-one/blob/main/community-containers/plex/plex.json or the one in Memories container)

I'm wondering what the idea is behind this...

By mounting the data dir, will all files of all users be susceptible to being served by the media server?
Can it be somehow restricted to certain users/folders or - even better - limited to some external storage / shared folders?

How is the access to these file set up so that e.g. plex can read them but not alter/delete them?

I personally wouldn't want to have all media files served by the media server. I know that having the DATADIR as mountpoint does not necessarily mean that it will be served by the respective media server but data not designated for media streaming should IMO not be accessible. 

Thanks in advance for explanations
Jochen

**Comment by szaimen** [comment, marked as answer]
> What the community container JSONs all ahve in common is the %NEXTCLOUD_DATADIR% as main mountpoint for media. (see https://github.com/nextcloud/all-in-one/blob/main/community-containers/plex/plex.json or the one in Memories container)
> 
> I'm wondering what the idea is behind this...

The idea is that Plex can have access to the datadir so that you can for example view your photos inside Plex.

> By mounting the data dir, will all files of all users be susceptible to being served by the media server? 

Yes

> Can it be somehow restricted to certain users/folders or - even better - limited to some external storage / shared folders?

No

> How is the access to these file set up so that e.g. plex can read them but not alter/delete them?

The mount is read-only: https://github.com/nextcloud/all-in-one/blob/8bf71c1052dedd774a5e86616ad8590b879e1e12/community-containers/plex/plex.json#L23-L25
